### PR TITLE
Mitigate detached head checkout

### PR DESCRIPTION
**Pull request by surajrav**
This makes it so that there is no special case of a commit hash based checkout in the freenas10 build.

This also allows SHALLOW GIT CHECKOUTS feature to work again.

Tested to build and boot up as of yesterday night (14th Jan, 2017)

### CODE REVIEW

File: build/profiles/freenas10/ports-middleware.pyd [hunk 1]
```diff
@@ -290,7 +281,7 @@ ports += {
 dedicated_repo_ports = [
     "py-freenas.cli", "py-freenas.utils", "py-libzfs", "py-netif", "py-cam", "py-SMART", "py-ws4py", "py-dhcp",
     "py-bsd", "py-smbconf", "py-pf", "py-wbclient", "py-krb5", "py-dockerhub", "licenselib", "py-ipfs-api",
-    "py-msock", "freenas-pkgtools", "py-filewrap"
+    "py-msock", "freenas-pkgtools", "py-filewrap", "freenas-migrate93"
```
Comment (chris3torek) [inline build/profiles/freenas10/ports-middleware.pyd:284]: I just poked around on github to find freenas-migrate93, it's in `freenas/migrate93` not `freenas/freenas-migrate93`. Is this line wrong or is the existing name of the repo wrong? They can't both be right...?

Comment (jceel) [reply build/profiles/freenas10/ports-middleware.pyd:284]: This is the port name, not repo name

Comment (chris3torek) [reply build/profiles/freenas10/ports-middleware.pyd:284]: Aha.  How does the repo-to-port-name mapping work then?

Comment (jceel) [reply build/profiles/freenas10/ports-middleware.pyd:284]: Port name must match "name" key in repos.py entries

Comment (jceel) [reply build/profiles/freenas10/ports-middleware.pyd:284]: ...but it doesn't!

Comment (surajrav) [reply build/profiles/freenas10/ports-middleware.pyd:284]: @chris3torek thanks for pointing that out I am fixing it now

**Pull request merged**